src/main/java/de/tum/cit/aet/artemis/buildagent/service/SharedQueueProcessingService.java

```diff
@@ -82,13 +89,26 @@ public class SharedQueueProcessingService {
 
     private UUID listenerId;
 
+    /**
+     * Scheduled future for checking availability and processing next build job.
+     */
+    private ScheduledFuture<?> scheduledFuture;
+
+    private volatile boolean isPaused = false;
+
+    private volatile boolean processResults = true;
```

Comment:
What is the difference between these two attributes?

Reply:
processResults is set after the grace period is over. This is necessary to make sure that build jobs that ended during the build jobs produce a result